pkg/install/error.go

```diff
@@ -0,0 +1,37 @@
+package install
+
+import (
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// isResourceQuotaExceededError returns true if it is Quota error with the
+// original error message
+func isResourceQuotaExceededError(err error) (bool, string) {
+	if detailedErr, ok := err.(autorest.DetailedError); ok {
+		// error format:
+		// (autorest.DetailedError).Original.(*azure.ServiceError).Details.([]map[string]interface{})
+		if serviceErr, ok := detailedErr.Original.(*azure.ServiceError); ok {
+			for _, d := range serviceErr.Details {
+				if d["code"].(string) == api.CloudErrorCodeQuotaExceeded {
+					return true, d["message"].(string)
+				}
```

Comment:
I think `if code, ok := d["code"].(string); ok` is needed - and for message - protect ourselves against a panic